### PR TITLE
added axios to dependencies and also added axios to serach component

### PR DESCRIPTION
**Pull request by marcinel123**
Hi Adam,

Please have a look as I already added axios functionality to the Search component.

Without any given Ip address it should show you your public ip and it works fine at my side.

Regards,
Martin

### CODE REVIEW

File: src/Components/Search.tsx [hunk 1]
```diff
@@ -1,4 +1,27 @@
+import React, { useEffect, useState } from "react";
+import api_baseUrl from "../api/api_baseUrl";
+
 export const Search = () => {
+	const [ipAddress, setIpAddress] = useState("");
+
+	const apiKey: string = "at_5I9sUYamw5MsQycwQTbrB6gMIRezB";
```
Comment (AdamKniec) [inline src/Components/Search.tsx]: You made a big mistake here. This is something called a secret. This should not be publically available. Create an .env file and store this value there. Let me know If You need help with that

File: src/Components/Search.tsx [hunk 2]
```diff
@@ -1,4 +1,27 @@
...
+
+	useEffect(() => {
+		const fetchLocation = async () => {
```
Comment (AdamKniec) [inline src/Components/Search.tsx]: move the fetch location function to a separate function to a separate folder called API

Comment (marcinel123) [reply src/Components/Search.tsx]: done

File: src/Components/Search.tsx (hunk 1 above)
Comment (marcinel123) [reply src/Components/Search.tsx]: changed

File: public/env.js [hunk 3]
```diff
@@ -0,0 +1 @@
+window.env = {};
```
Comment (AdamKniec) [inline public/env.js]: what's that? :D

File: src/Components/Search.tsx [hunk 4]
```diff
@@ -1,4 +1,17 @@
+import React, { useEffect, useState } from "react";
+import { useFetchLocation } from "../api/apiFetchLocation";
+
 export const Search = () => {
+	const [ipAddress, setIpAddress] = useState("209.142.68.29");
+	const { fetchLocation } = useFetchLocation(ipAddress);
+
+	console.log(process.env.REACT_APP_KEY);
```
Comment (AdamKniec) [inline src/Components/Search.tsx]: remove the console logs

File: src/api/apiFetchLocation.tsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import api_baseUrl from "./api_baseUrl";
+
+export const useFetchLocation = (ipAddress: string) => {
+	const fetchLocation = async () => {
+		try {
+			const response = await api_baseUrl.get(
+				`/country?apiKey=${process.env.REACT_APP_KEY}&ipAddress=${ipAddress}`
+			);
+			console.log(response.data);
```
Comment (AdamKniec) [inline src/api/apiFetchLocation.tsx]: remove console log

File: src/api/apiFetchLocation.tsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+		} catch (err: any) {
```
Comment (AdamKniec) [inline src/api/apiFetchLocation.tsx]: any is not a proper type. Change to unknown

File: public/index.html [hunk 7]
```diff
@@ -25,6 +25,7 @@
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
     <title>React App</title>
+    <script id="react-dotenv" src="./env.js"></script>
```
Comment (AdamKniec) [inline public/index.html]: why is that needed?

File: public/env.js (hunk 3 above)
Comment (AdamKniec) [reply public/env.js]: this file should be called .env and should be moved to the root directory.

Comment (marcinel123) [reply public/env.js]: I have not seen this before. I removed the file. It was not added by me. Probably it added automatically when I installed one of env libraries. Fixed.

File: src/Components/Search.tsx (hunk 4 above)
Comment (marcinel123) [reply src/Components/Search.tsx]: fixed.

File: src/api/apiFetchLocation.tsx (hunk 5 above)
Comment (marcinel123) [reply src/api/apiFetchLocation.tsx]: fixed.

File: src/api/apiFetchLocation.tsx (hunk 6 above)
Comment (marcinel123) [reply src/api/apiFetchLocation.tsx]: changed.

File: public/index.html (hunk 7 above)
Comment (marcinel123) [reply public/index.html]: removed. I has been left after uninstalling env libraries.

**Review by AdamKniec — APPROVED** [review]
